bp-image.html

```diff
@@ -144,48 +183,133 @@
+        </button>
+        <button id="flip-button" type="submit">
+          <i class="fa-solid fa-arrows-up-down"></i>
+        </button>
+      </div>
       <div class="canvas-container">
         <canvas id="myCanvas" width="500" height="500"></canvas>
       </div>
     </div>
   </div>
 
+  <script>
+    const rotateButton = document.getElementById('rotate-button');
+    rotateButton.addEventListener('click', async () => {
+      const form = document.querySelector('form');
+      const formData = new FormData(form);
+      formData.append('rotate', 1);
+      const urlParams = new URLSearchParams();
+      for (const [name, value] of formData.entries()) {
+        urlParams.set(name, value);
+      }
+      const newUrl = `${window.location.pathname}?${urlParams.toString()}`;
+      window.location.href = newUrl;
+
+    });
+
+    const flipButton = document.getElementById('flip-button');
+    flipButton.addEventListener('click', () => {
+      const form = document.querySelector('form');
+      const formData = new FormData(form);
+      formData.append('flip', 1);
+      const urlParams = new URLSearchParams();
+      for (const [name, value] of formData.entries()) {
+        urlParams.set(name, value);
+      }
+      const newUrl = `${window.location.pathname}?${urlParams.toString()}`;
+      window.location.href = newUrl;
+    });
+
+    function openLightbox() {
+      const canvasContainer = document.querySelector('.canvas-container');
+      const canvas = document.getElementById('myCanvas');
+
+      canvasContainer.classList.add('canvas-lightbox');
+      document.body.classList.add('no-scroll'); // Add this line
+
+      const canvasBox = document.createElement('div');
+      canvasBox.classList.add('canvas-box');
+      canvasContainer.appendChild(canvasBox);
+      canvasBox.appendChild(canvas);
+
+      canvasContainer.addEventListener('click', closeLightbox);
+    }
+
+    function closeLightbox() {
+      const canvasContainer = document.querySelector('.canvas-container');
+      const canvas = document.getElementById('myCanvas');
+
+      canvasContainer.classList.remove('canvas-lightbox');
+      canvas.classList.remove('canvas-lightbox');
+
+      canvas.removeEventListener('click', closeLightbox);
+    }
+
+    const lightboxBtn = document.getElementById('open-lightbox');
+    lightboxBtn.addEventListener('click', openLightbox);
+  </script>
+
   <script type="module">
     import { InteractiveBlueprint } from "./blueprint-interactive.js";
+    import { Blueprint } from "./blueprint.js";
     import { styles, TRANSFORMATIONS } from "./constants.js";
     const canvas = document.getElementById('myCanvas');
 
     const queryParams = new URLSearchParams(window.location.search);
-    const bps = queryParams.get('bp');
-    
-    // Size of HTML document (same as pageHeight/pageWidth in screenshot).
+    var bps = queryParams.get('bp');
+    const rotate = queryParams.get('rotate', false) ?? false;
```

Comment:
Where is rotate counter-clockwise and horizontal flip?

Reply:
I figure they can just rotate multiple times to get what they want